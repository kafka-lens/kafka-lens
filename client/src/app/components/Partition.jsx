import React from 'react';
import '../css/Partition.scss';

const Partition = ({ id, showMessages, topicName }) => (
  <div
    id={id}
    className="single-partition"
    role="button"
    tabIndex="0"
    onKeyPress={showMessages}
    onClick={showMessages}
    topicname={topicName}
  >
    {`Partition  ${id}`}
  </div>
);

export default Partition;
