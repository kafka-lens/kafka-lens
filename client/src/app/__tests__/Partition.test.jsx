import React from 'react';
import { shallow } from 'enzyme';

import Partition from '../components/Partition';

describe('Parition.js unit tests', () => {
  let wrapper;
  const props = {
    id: '5',
    showMessages: jest.fn(),
    topicName: 'Test',
  };

  beforeAll(() => {
    wrapper = shallow(<Partition {...props} />);
  });

  it('Should render without errors', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('Should render a div with inner text equal to Parition', () => {
    expect(wrapper.find('div').text()).toEqual('Partition  5');
  });

  it('Should call showMessages on click', () => {
    wrapper
      .find('div')
      .at(0)
      .simulate('click');
    expect(props.showMessages).toHaveBeenCalled();
  });
});
