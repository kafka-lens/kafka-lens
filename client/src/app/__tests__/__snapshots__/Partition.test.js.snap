// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Parition.js unit tests Should render without errors 1`] = `
<div
  className="single-partition"
  id="5"
  onClick={[MockFunction]}
  onKeyPress={[MockFunction]}
  role="button"
  tabIndex="0"
  topicname="Test"
>
  Partition  5
</div>
`;
